packages/common/src/adapters/track.ts

```diff
@@ -219,3 +222,119 @@ export const stemTrackMetadataFromSDK = (
     is_owned_by_user: false
   }
 }
+
+export const trackMetadataForUploadToSdk = (
+  input: TrackMetadataForUpload
+): TrackMetadata => ({
+  ...camelcaseKeys(
+    pick(input, [
+      'title',
+      'description',
+      'license',
+      'isrc',
+      'iswc',
+      'genre',
+      'mood',
+      'tags',
+      'is_unlisted',
+      'field_visibility',
+      'is_premium',
+      'premium_conditions',
+      'is_stream_gated',
+      'stream_conditions',
+      'is_download_gated',
+      'download_conditions',
+      'orig_file_cid',
+      'orig_filename',
+      'is_downloadable',
+      'is_original_available',
+      'bpm',
+      'is_custom_bpm',
+      'musical_key',
+      'is_custom_musical_key',
+      'comments_disabled',
+      'ddex_release_ids',
+      'parental_warning_type',
+      'audio_upload_id',
+      'ai_attribution_user_id'
+    ])
+  ),
+  title: input.title,
+  description: input.description ?? undefined,
+  mood: input.mood as Mood,
+  tags: input.tags ?? undefined,
+  genre: (input.genre as Genre) || undefined,
+  releaseDate: input.release_date ? new Date(input.release_date) : undefined,
+  previewStartSeconds: input.preview_start_seconds ?? undefined,
+  previewCid: input.preview_cid ?? '',
+  ddexApp: input.ddex_app ?? '',
+  aiAttributionUserId: input.ai_attribution_user_id
+    ? encodeHashId(input.ai_attribution_user_id)
```

Comment:
Would again recommend `OptionalId` here as I think it will accept null/undefined as input and give you back undefined if it's not a valid id.
Or you can do `OptionalId(input.ai_attribution_user_id) ?? undefined`